app/controllers/games_controller.rb

```diff
@@ -17,19 +17,25 @@ def show
     @game = Game.find(params[:id])
   end
 
+  def join
+    @game = Game.find(params[:id]).available
+    @game.update_attributes(@game.black_player_id = current_user.id)
```

Comment:
In your call to update_attributes method, this can only accept a Hash argument. The current implementation throws an error when it tries to update.  Try this instead:

`@game.update_attributes(black_player_id: current_user.id)`